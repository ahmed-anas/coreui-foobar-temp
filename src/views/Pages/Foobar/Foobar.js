import React, { Component } from 'react';

class Foobar extends Component {
    render() {
        return (
            <div className="app flex-row align-items-center">
                this is foobar!!yaiiyad
 
                
          <input 
                    type="text"
                    name="mainOther" />
          <input
                    onChange={(event) => { this.setState({ input: event.target.value }) }}
                    type="text"
                    name="main" />
                    
          <input
                    type="text"
                    name="mainMan" />
            </div>
        );
    }
}

export default Foobar;
