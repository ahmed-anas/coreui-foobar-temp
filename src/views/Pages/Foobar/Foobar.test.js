import React from 'react';
import { shallow } from "enzyme";
import { isIterable } from 'core-js';
import Foobar from './Foobar';

describe('Foobar', ()=>{
    it('should render properly', ()=>{
        const wrapper = shallow(<Foobar />)
    });

    it('it should match snap', ()=>{
        const wrapper = shallow(<Foobar />)
        expect(wrapper.getElements()).toMatchSnapshot();
    })

    it('should properly handle input change', ()=>{
        const component = shallow(<Foobar />)
        const form = component.find("input[name='main']");

        form.props().onChange({target: {
            name: 'random doesn\'t matter',
            value: 'I am the man'
        }});

        expect(component.state('input')).toEqual('I am the man')
        
        

    })
})