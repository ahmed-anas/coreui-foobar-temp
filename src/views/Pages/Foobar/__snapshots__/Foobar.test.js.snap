// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Foobar it should match snap 1`] = `
Array [
  <div
    className="app flex-row align-items-center"
  >
    this is foobar!!yaiiyad
    <input
      name="mainOther"
      type="text"
    />
    <input
      name="main"
      onChange={[Function]}
      type="text"
    />
    <input
      name="mainMan"
      type="text"
    />
  </div>,
]
`;
